homeassistant/components/light/xiaomi_miio.py

```diff
@@ -452,8 +595,229 @@ def max_mireds(self):
         """Return the warmest color_temp that this light supports."""
         return 370
 
+    @property
+    def supported_features(self):
+        """Return the supported features."""
+        return SUPPORT_FLAGS_CEILING
+
+    @asyncio.coroutine
+    def async_update(self):
+        """Fetch state from the device."""
+        from miio import DeviceException
+        try:
+            state = yield from self.hass.async_add_job(self._light.status)
+            _LOGGER.debug("Got new state: %s", state)
+
+            self._state = state.is_on
+            self._brightness = ceil((255 / 100.0) * state.brightness)
+            self._color_temp = self.translate(
+                state.color_temperature,
+                CCT_MIN, CCT_MAX,
+                self.max_mireds, self.min_mireds)
+
+            delayed_turn_off = self.delayed_turn_off_timestamp(
+                state.delay_off_countdown,
+                dt.utcnow(),
+                self._state_attrs[ATTR_DELAYED_TURN_OFF])
+
+            self._state_attrs.update({
+                ATTR_SCENE: state.scene,
+                ATTR_DELAYED_TURN_OFF: delayed_turn_off,
+                ATTR_NIGHT_LIGHT_MODE: state.smart_night_light,
+                ATTR_AUTOMATIC_COLOR_TEMPERATURE:
+                    state.automatic_color_temperature,
+            })
+
+        except DeviceException as ex:
+            self._state = None
+            _LOGGER.error("Got exception while fetching the state: %s", ex)
+
 
-class XiaomiPhilipsEyecareLamp(XiaomiPhilipsGenericLight, Light):
+class XiaomiPhilipsEyecareLamp(XiaomiPhilipsGenericLight):
     """Representation of a Xiaomi Philips Eyecare Lamp 2."""
 
-    pass
+    def __init__(self, name, light, model):
+        """Initialize the plug switch."""
+        XiaomiPhilipsGenericLight.__init__(self, name, light, model)
+
+        self._state_attrs.update({
+            ATTR_REMINDER: None,
+            ATTR_NIGHT_LIGHT_MODE: None,
+            ATTR_EYECARE_MODE: None,
+        })
+
+    @property
+    def supported_features(self):
+        """Return the supported features."""
+        return SUPPORT_FLAGS_SREAD1_EYECARE_LIGHT
+
+    @asyncio.coroutine
+    def async_update(self):
+        """Fetch state from the device."""
+        from miio import DeviceException
+        try:
+            state = yield from self.hass.async_add_job(self._light.status)
```

Comment:
If you're going to call a sync method, you might as well just make the whole update method not async